lib/view_model/anon_invoice_page_view_model.dart

```diff
@@ -0,0 +1,192 @@
+        cryptoCurrency = walletTypeToCryptoCurrency(_wallet.type) {
+    _getPreviousDonationLink();
+    _fetchLimits();
+  }
+
+  List<Currency> get currencies => [walletTypeToCryptoCurrency(_wallet.type), ...FiatCurrency.all];
+  final AnonPayApi anonPayApi;
+  final String address;
+  final SettingsStore settingsStore;
+  final WalletBase _wallet;
+  final Box<AnonpayInvoiceInfo> _anonpayInvoiceInfoSource;
+  final FiatConversionStore _fiatConversationStore;
+  final SharedPreferences sharedPreferences;
+  final ReceivePageOption pageOption;
+
+  @observable
+  Currency selectedCurrency;
+
+  CryptoCurrency cryptoCurrency;
+
+  @observable
+  String receipientEmail;
+
+  @observable
+  String receipientName;
+
+  @observable
+  String description;
+
+  @observable
+  String amount;
+
+  @observable
+  ExecutionState state;
+
+  @computed
+  int get selectedCurrencyIndex => currencies.indexOf(selectedCurrency);
+
+  @observable
+  double? minimum;
+
+  @observable
+  double? maximum;
+
+  @action
+  void selectCurrency(Currency currency) {
+    selectedCurrency = currency;
+    maximum = minimum = null;
+    if (currency is CryptoCurrency) {
+      cryptoCurrency = currency;
+    }
+    _fetchLimits();
+  }
+
+  @action
+  Future<void> createInvoice() async {
+    state = IsExecutingState();
+    if (amount.isNotEmpty) {
+      final amountInCrypto = double.parse(amount);
+      if (minimum != null && amountInCrypto < minimum!) {
+        state = FailureState('Amount is too small');
+        return;
+      }
+      if (maximum != null && amountInCrypto > maximum!) {
+        state = FailureState('Amount is too big');
+        return;
+      }
+    }
+    final result = await anonPayApi.createInvoice(AnonPayRequest(
+      cryptoCurrency: cryptoCurrency,
+      address: address,
+      amount: amount.isEmpty ? null : amount,
+      description: description,
+      email: receipientEmail,
+      name: receipientName,
+      fiatEquivalent:
+          selectedCurrency is FiatCurrency ? (selectedCurrency as FiatCurrency).raw : null,
+    ));
+
+    _anonpayInvoiceInfoSource.add(result);
+
+    state = ExecutedSuccessfullyState(payload: result);
+  }
+
+  @action
+  void setRequestParams({
+    required String inputAmount,
+    required String inputName,
+    required String inputEmail,
+    required String inputDescription,
+  }) {
+    receipientName = inputName;
+    receipientEmail = inputEmail;
+    description = inputDescription;
+    amount = inputAmount;
+  }
+
+  @action
+  Future<void> generateDonationLink() async {
+    state = IsExecutingState();
+
+    final result = await anonPayApi.generateDonationLink(AnonPayRequest(
+      cryptoCurrency: cryptoCurrency,
+      address: address,
+      description: description,
+      email: receipientEmail,
+      name: receipientName,
+    ));
+
+    await sharedPreferences.setString(PreferencesKey.trocadorDonationLink, result.clearnetUrl);
```

Comment:
I would suggest saving the whole result thus you don't have to call the API the next time the user goes here, causing the app to show the form for a couple of seconds then navigate to the QR page

Reply:
then you can check for the saved value before even navigating to the donation form